### PR TITLE
Add task retries smoke product test 

### PR DESCRIPTION
**Pull request by losipiuk**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by martint** [comment]
Why does this need to be a product test instead of a regular test using the distributed query runner and multiple in-process nodes?

**Comment by losipiuk** [comment]
> Why does this need to be a product test instead of a regular test using the distributed query runner and multiple in-process nodes?

Regular test did not catch jackson serialization problems introduced with https://github.com/trinodb/trino/pull/22722 (then reverted).
I do not fully understand the root cause - but classloader setup is different in DQR based tests than what we have in actual deployment - so I guess this is the problem.

DQR queries run fine but with product tests we get:

```2024-07-24T20:03:59.899+0545    WARN    http-worker-71  io.trino.server.ThrowableMapper Request failed for /v1/task/20240724_141842_00000_hzujt.0.0.4
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class io.trino.plugin.base.metrics.TDigest and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: io.tr
ino.execution.TaskInfo["stats"]->io.trino.operator.TaskStats["pipelines"]->com.google.common.collect.RegularImmutableList[0]->io.trino.operator.PipelineStats["operatorSummaries"]->com.google.common.collect.RegularImmutableList[0]->io.trino.operator.OperatorStats["pipelineMetrics"]->jav
a.util.ImmutableCollections$MapN["FileSystemExchangeSource.s3GetObjectRequestsFailed"]->io.trino.plugin.base.metrics.TDigestHistogram["digest"])
        at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
        at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1330)
        at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:414)
        at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:53)
        at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:30)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:653)
        at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeTypedFields(MapSerializer.java:1027)
        at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields(MapSerializer.java:779)
        at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo(MapSerializer.java:764)
        at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:720)
        at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:35)
        at com.fasterxml.jackson.databind.ser.std.JsonValueSerializer.serialize(JsonValueSerializer.java:289)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
```

on worker with https://github.com/trinodb/trino/pull/22722 applied.

**Comment by losipiuk** [comment]
> > Why does this need to be a product test instead of a regular test using the distributed query runner and multiple in-process nodes?
> 
> Regular test did not catch jackson serialization problems introduced with #22722 (then reverted). I do not fully understand the root cause - but classloader setup is different in DQR based tests than what we have in actual deployment - so I guess this is the problem.
> ....

https://github.com/trinodb/trino/pull/22797 fixes the problem - but it would be great to have regression test coverage.

**Comment by wendigo** [comment]
I don't think that adding a product test is worth it tbh. I'd prefer to create some test that basically creates a plugin class loader and tests whether you can serialize/deserialize class when you are in the server class loader.

**Comment by losipiuk** [comment]
> I don't think that adding a product test is worth it tbh. I'd prefer to create some test that basically creates a plugin class loader and tests whether you can serialize/deserialize class when you are in the server class loader.

How would that be different from test which uses DistributedQueryRunner - where serialization works fine for some reason? This PT is extra 2 mins of CI runtime (in a job which is short anyway so does not extend overall CI for PR), so I do not think it is a big deal.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeMinioDataLakeTaskRetriesFilesystem.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        builder.configureContainer(TESTS, dockerContainer -> {
+            dockerContainer.withEnv("S3_BUCKET", S3_BUCKET_NAME);
+        });
+
+        // initialize buckets in minio
+        FileAttribute<Set<PosixFilePermission>> posixFilePermissions = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"));
+        Path minioBucketDirectory;
+        try {
+            minioBucketDirectory = Files.createTempDirectory("test-bucket-contents", posixFilePermissions);
```
Comment (wendigo) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeMinioDataLakeTaskRetriesFilesystem.java:70]: Do we need to expose minio data to the host?

**Review by wendigo — APPROVED** [review]

**Pull request merged**